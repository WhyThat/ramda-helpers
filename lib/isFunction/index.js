import { is } from 'ramda';

/**
 * Returns true if prop is a function
 * @name isFunction
 * @param {*} value
 * @returns {Boolean}
 * @example
 * isFunction({}) //=> false
 * isFunction(() => {}) //=> true
 */
const isFunction = is(Function);
export default isFunction;
