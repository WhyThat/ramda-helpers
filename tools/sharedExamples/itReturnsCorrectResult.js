export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
