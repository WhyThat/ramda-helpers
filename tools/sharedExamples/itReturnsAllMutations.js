import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
