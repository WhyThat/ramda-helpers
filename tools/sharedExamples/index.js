
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
