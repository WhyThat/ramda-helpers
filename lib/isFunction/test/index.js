import { itReturnsAllMutations } from '../../../tools/sharedExamples';
import isFunction from '../';

describe('isFunction', () => {
  const mutations = [{
    name: 'With null',
    props: null,
  }, {
    name: 'With array',
    props: [],
  }, {
    name: 'With object',
    props: {},
  }, {
    name: 'With function',
    props: () => {},
  }];

  itReturnsAllMutations(isFunction, mutations);
});
