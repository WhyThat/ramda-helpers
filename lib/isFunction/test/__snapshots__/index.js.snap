exports[`isFunction With array returns correct result 1`] = `false`;

exports[`isFunction With function returns correct result 1`] = `true`;

exports[`isFunction With null returns correct result 1`] = `false`;

exports[`isFunction With object returns correct result 1`] = `false`;
